lib/src/migrators/division.dart

```diff
@@ -244,40 +272,45 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
       return false;
     }
 
-    ListExpression? channels;
-    if (node.arguments.positional.length == 1 &&
-        node.arguments.named.isEmpty &&
-        node.arguments.positional.first is ListExpression) {
-      channels = node.arguments.positional.first as ListExpression?;
-    } else if (node.arguments.positional.isEmpty &&
-        node.arguments.named.containsKey(r'$channels') &&
-        node.arguments.named.length == 1 &&
-        node.arguments.named[r'$channels'] is ListExpression) {
-      channels = node.arguments.named[r'$channels'] as ListExpression?;
-    }
-    if (channels == null ||
-        channels.hasBrackets ||
-        channels.separator != ListSeparator.space ||
-        channels.contents.length != 3 ||
-        channels.contents.last is! BinaryOperationExpression) {
-      return false;
-    }
+    var channels = switch (node.arguments) {
+      ArgumentInvocation(positional: [ListExpression arg, ...], named: Map()) =>
```

Comment:
Should this be `[ListExpression arg]`? The old code only matches if there's only one positional argument.

Also, I think `named: Map()` just checks that `named`'s type is `Map`. If you want to check that it's empty, you'll need to do `named: Map(length: 0)`.

Reply:
Fixed